File: 백준/Silver/10828. 스택/스택.cc
#include<iostream>
#include<stack>

using namespace std;


int main(){
    ios_base::sync_with_stdio(false);

    int N = 0;
    string cmd;
    int value = 0;
    stack<int> block;

    cin >> N;

    for(int i = 0; i<N; i++){
        cin >> cmd;
        if(cmd == "push"){
            cin >> value;
            block.push(value);
        }
        else if(cmd == "pop" && !block.empty()){
            printf("%d\n", block.top());
            block.pop();
        }
        else if(cmd == "size") printf("%d\n", block.size());
        else if(cmd == "empty") printf("%d\n", block.empty());
        else if(cmd == "top" && !block.empty()) printf("%d\n", block.top());
        else printf("-1\n");
    }

    return 0;
}

File: 백준/Silver/10828. 스택/스택.py
#Silver IV  
import sys
input = sys.stdin.readline
N = int(input())
cmd = []
stack = []

for _ in range(N):
    value = -1
    cmd = list(input().split())

    if cmd[0] == 'push':
        stack.append(cmd[1])
    else:
        match (cmd[0]):
            case 'top':
                if not stack: print(-1)
                else: print(stack[-1])
            case 'size':
                print(len(stack))
            case 'empty':
                if not stack: print(1)
                else: print(0)
            case 'pop':
                if not stack: print(-1)
                else:
                    print(stack[-1])
                    stack.pop()

